vortex-datetime-dtype/src/arrow.rs

```diff
@@ -44,12 +44,44 @@ pub fn make_temporal_ext_dtype(data_type: &DataType) -> ExtDType {
             DATE_ID.clone(),
             Some(TemporalMetadata::Date(TimeUnit::Ms).into()),
         ),
-        _ => unimplemented!("we should fix this"),
+        _ => unimplemented!("{data_type} conversion"),
+    }
+}
+
+/// Convert temporal ExtDType to a corresponding arrow DataType
```

Comment:
we should document that this will panic if the ExtDType is not one of the temporal DTypes